cargo-workspaces/src/publish.rs

```diff
@@ -100,7 +101,10 @@ impl Publish {
                     warn!("build failed", "");
                 }
 
-                basic_checks(pkg)?;
+                command_success = command_success && basic_checks(pkg)?;
+                // We cannot use `cargo publish --dry-run` directly, because it checks if all the dependencies
```

Comment:
I kept the basic checks, but I was wondering if we still need them if `--dry-run` warns about all of the issues that we check here?

Reply:
No, `--dry-run` doesn't warn about them -- they are only checked during uploading.
Probably it makes sense to add this functionality to `cargo` directly, I'll create an issue there.

Reply:
Yeah, that's probably the way we want to go in the future.